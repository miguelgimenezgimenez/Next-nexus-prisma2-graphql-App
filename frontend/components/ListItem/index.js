import React from 'react';
import Link from 'next/link';
import { Title } from '../styles/GlobalStyles';
import ListItemStyles from './styles';

export default function ListItem({ item, pathname }) {

  return (
    <ListItemStyles>

      <Title>
        <Link
          href={{
            pathname: pathname,
            query: { id: item.id },
          }}
        >
          <a>
            {item.image && (
              <img src={item.image} alt={item.name} />
            )}
            {item.name}</a>
        </Link>
      </Title>

      <p>{item.description}</p>

      <div className="buttonList">
        <Link
          href={{
            pathname: 'update',
            query: { id: item.id },
          }}
        >
          <a>Edit</a>
        </Link>

      </div>
    </ListItemStyles>
  );
}
