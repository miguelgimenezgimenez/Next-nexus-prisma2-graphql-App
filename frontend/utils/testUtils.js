import casual from 'casual';


casual.seed(777);
const mockBrand = () => ({
  id: 1,
  name: casual.name
});

const mockPhone = overrides => ({
  "id": 2,
  "name": casual.name,
  "image":"image.jpg",
  "brand_id": 1,
  "storage": "16GB/32GB storage, microSDXC",
  "dimensions": "260g, 9.4mm thickness",
  "os": "Android 6.0",
  ...overrides
});





export {
  mockBrand,
  mockPhone,
};
