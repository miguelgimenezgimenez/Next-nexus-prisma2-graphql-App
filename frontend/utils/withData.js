import { ApolloClient, ApolloLink, InMemoryCache } from '@apollo/client';
import { onError } from '@apollo/link-error';
import { getDataFromTree } from '@apollo/react-ssr';
import withApollo from 'next-with-apollo';
import { endpoint } from '../config';

import { createHttpLink } from "apollo-link-http";


function createClient({ initialState }) {

  return new ApolloClient({
    link: ApolloLink.from([
      onError(({ graphQLErrors, networkError }) => {
        if (graphQLErrors)
          graphQLErrors.forEach(({ message, locations, path }) =>
            console.log(
              `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`
            )
          );
        if (networkError)
          console.log(
            `[Network error]: ${networkError}. Backend is unreachable. Is it running?`
          );
      }),

      createHttpLink({ uri: endpoint })

    ]),

    cache: new InMemoryCache({}).restore(initialState || {}),
  });
}

export default withApollo(createClient, { getDataFromTree });