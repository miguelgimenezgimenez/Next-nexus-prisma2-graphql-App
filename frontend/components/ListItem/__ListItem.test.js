import { render, screen } from '@testing-library/react'
import { MockedProvider } from '@apollo/react-testing'
import ListItem from './index'
import { mockPhone } from '../../utils/testUtils'

const phone = mockPhone()


describe('<ListItem/>', () => {
  it('renders and matches the snapshot', () => {
    const { container } = render(
      <MockedProvider>
        <ListItem item={phone} pathname={'/phone'} />
      </MockedProvider>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders the image properly', () => {
    render(
      <MockedProvider>
        <ListItem item={phone} />
      </MockedProvider>
    )
    const img = screen.getByAltText(phone.name)
    expect(img).toBeInTheDocument()
  })

  it('renders the name', () => {
    const { container } = render(
      <MockedProvider>
        <ListItem item={phone} pathname="/phone" />
      </MockedProvider>
    )
    expect(screen.getByText(`${phone.name}`)).toBeInTheDocument()
    const link = container.querySelector('a')
    expect(link).toHaveAttribute('href', `/phone?id=${phone.id}`)
    expect(link).toHaveTextContent(phone.name)
  })

  it('renders out the buttons properly', () => {
    render(
      <MockedProvider>
        <ListItem item={phone} />
      </MockedProvider>
    )

    const edit = screen.getByText(/Edit/i)
    expect(edit.href).toContain(`/update?id=${phone.id}`)

    const deleteItem = screen.getByText(/delete/i)
    expect(deleteItem).toHaveProperty('type', 'button')
    expect(deleteItem).toBeInTheDocument()
  })
})
