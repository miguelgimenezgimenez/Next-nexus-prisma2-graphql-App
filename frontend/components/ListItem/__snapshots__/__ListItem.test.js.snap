// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem/> renders and matches the snapshot 1`] = `
<div>
  <div
    class="styles__ListItemStyles-yahwwc-0 iUlpUq"
  >
    <h3
      class="common__Title-qhs55p-0 euMuWN"
    >
      <a
        href="/phone?id=2"
      >
        <img
          alt="Miss Coleman Berge"
          src="Ms. Robin Luettgen"
        />
        Miss Coleman Berge
      </a>
    </h3>
    <div
      class="buttonList"
    >
      <a
        href="/update?id=2"
      >
        Edit
      </a>
      <button
        type="button"
      >
        Delete Phone
      </button>
    </div>
  </div>
</div>
`;
